bouncer/src/repo/manipulator/modelconvertor/import/odaHelper/geometry_collector.cpp

```diff
@@ -79,94 +81,203 @@ repo::core::model::RepoNodeSet repo::manipulator::modelconvertor::odaHelper::Geo
+				std::pair<repo::lib::RepoVector3D64,
+				std::pair<int, repo::lib::RepoVector3D64>>(
+					v,
+					std::pair<int,
+					repo::lib::RepoVector3D64>(
+						vertIdx,
+						repo::lib::RepoVector3D64())));
+
+
+			// if the vertex has not yet been seen, also update the bounding box
+
+			if (!meshData->boundingBox.size())
+			{
+				meshData->boundingBox.push_back({ (float)v.x, (float)v.y, (float)v.z });
+				meshData->boundingBox.push_back({ (float)v.x, (float)v.y, (float)v.z });
+			}
+			else
+			{
+				meshData->boundingBox[0][0] = meshData->boundingBox[0][0] > v.x ? (float)v.x : meshData->boundingBox[0][0];
+				meshData->boundingBox[0][1] = meshData->boundingBox[0][1] > v.y ? (float)v.y : meshData->boundingBox[0][1];
+				meshData->boundingBox[0][2] = meshData->boundingBox[0][2] > v.z ? (float)v.z : meshData->boundingBox[0][2];
+
+				meshData->boundingBox[1][0] = meshData->boundingBox[1][0] < v.x ? (float)v.x : meshData->boundingBox[1][0];
+				meshData->boundingBox[1][1] = meshData->boundingBox[1][1] < v.y ? (float)v.y : meshData->boundingBox[1][1];
+				meshData->boundingBox[1][2] = meshData->boundingBox[1][2] < v.z ? (float)v.z : meshData->boundingBox[1][2];
+			}
+
+			if (minMeshBox.size()) {
+				minMeshBox[0] = v.x < minMeshBox[0] ? v.x : minMeshBox[0];
+				minMeshBox[1] = v.y < minMeshBox[1] ? v.y : minMeshBox[1];
+				minMeshBox[2] = v.z < minMeshBox[2] ? v.z : minMeshBox[2];
+			}
+			else {
+				minMeshBox = { v.x, v.y, v.z };
+			}
+		}
+		{
+			auto iterators = meshData->vToVIndex.equal_range(v);
+			vertIdx = iterators.first->second.first;
+		}
+
+		face.push_back(vertIdx);
+	}
+
+	meshData->faces.push_back(face);
 }
 
 void GeometryCollector::addFace(
 	const std::vector<repo::lib::RepoVector3D64> &vertices,
 	const repo::lib::RepoVector3D64& normal,
 	const std::vector<repo::lib::RepoVector2D>& uvCoords)
 {
-	if (!meshData.size()) startMeshEntry();
-
-	bool currentEntryHasUV = currentEntry->uvCoords.size();
-	bool faceHasUV = uvCoords.size();
-	if (currentEntry->rawVertices.size() && currentEntryHasUV != faceHasUV) {
-		//Start a new mesh if the current mesh has UVs but this face doesn't or vice versa.
-		auto lastMeshName = nextMeshName;
-		stopMeshEntry();
-		nextMeshName = lastMeshName + "_";
-		repoDebug << "Staring a new mesh due to UV difference";
-		startMeshEntry();
+	if (!vertices.size())
+	{
+		repoError << "Vertices size[" << vertices.size() << "] is unsupported. A face must have more than 0 vertices";
+		exit(-1);
 	}
 
-	if (faceHasUV && uvCoords.size() != vertices.size()) {
+	auto meshData = startOrContinueMeshByFormat(getMeshFormat(uvCoords.size(), true, vertices.size()));
+
+	if ((uvCoords.size() > 0) && uvCoords.size() != vertices.size()) {
 		repoError << "Vertices size[" << vertices.size() << "] and UV size [" << uvCoords.size() << "] mismatched!";
 		exit(-1);
 	}
 
+	// Todo: this method attempts to generate indices by matching vertex based on attributes, but doesn't yet match the uv coordinates.
```

Comment:
This doesn't break revit textured models right?

Would we be in a situation of joining up vertices with different UVs?

Reply:
@carmenfan Yes! This the bug we spoke about the other week. I've raised a maintenance issue here: https://github.com/3drepo/3drepobouncer/issues/432. The reason we won't have hit it yet is because to trigger it, there would have to be collocated vertices, with identical normals, but different UVs - that is, a UV map seam on a flush surface.
This would be a very very odd thing to do, because in that case one would just adjust the UV coordinates at the edges themselves to move, say, a decal, around.  
(Even in the case where one had something like a screen or poster on a wall it wouldn't be done, because only a single texture is applied per UV set, so changing the content would involve changing the UVs. Far easier to just to change the texture.)